openstack/blockstorage/extensions/quotasets/testing/requests_test.go

```diff
@@ -10,13 +11,20 @@ import (
 
 var emptyQuotaSet = quotasets.QuotaSet{}
 var emptyQuotaUsageSet = quotasets.QuotaUsageSet{}
+var emptyUpdateOpts = quotasets.UpdateOpts{}
 
-func testSuccessTestCase(t *testing.T, httpMethod, uriPath, jsonBody string, uriQueryParams map[string]string, expectedQuotaSet quotasets.QuotaSet, expectedQuotaUsageSet quotasets.QuotaUsageSet) error {
+func testSuccessTestCase(t *testing.T, httpMethod, uriPath, jsonBody string, uriQueryParams map[string]string, expectedQuotaSet quotasets.QuotaSet, expectedQuotaUsageSet quotasets.QuotaUsageSet, updateOpts quotasets.UpdateOpts) error {
```

Comment:
This might be getting a little unwieldy. I wonder if it would be more readable if each test case had its own function? I'm fine with either direction.

Reply:
I prefer this approach as it is (in my opinion) significantly better than the approach in the compute quotasets that I originally tried to copy. I found myself unable to really understand what was going on since it makes use of a nonsensical, inconsistent naming of fixture variables despite the ostensibly easier-to-understand breakdown of test cases into individual functions. I recall being convinced a couple weeks ago that the compute quotaset unit tests are not testing what the function names are claiming that they are testing because of erroneous re-use of fixture variables between test cases.

I could see breaking this function down into smaller functions and pushing the logic for which type of test case function needs to be invoked into `testSuccessTestCases`, but if you could go either way then I'd rather focus on higher priority work.

Reply:
I agree with @jtopjian : at the very least, this function has too many parameters. I'm also finding it difficult to track the fields of `successTestCases`. I agree the naming in the compute `quotasets` package could be improved, but I'd prefer if we kept the same testing format as in other packages.